### PR TITLE
🏗 Modernize / fix test logging during CI builds

### PR DESCRIPTION
**Pull request by rsimha**
During the migration to CircleCI, I discovered two problems:
- CircleCI doesn't wrap its logs, resulting in ~10k dots on one line during unit tests for example.
- New versions of Mocha have changed the reporter API, causing E2E test logging to break on CircleCI and [stall the job](https://app.circleci.com/pipelines/github/rsimha/amphtml/141/workflows/554d9f08-3e5a-4841-95b6-d29ca47561fc/jobs/539).

**This PR does two things:**
- Wraps all test logging to 150 dots per line
- Modernizes the custom reporter for e2e tests based on the latest mocha dev docs

**Passing result:** 

image

Also tested on CircleCI [here](https://app.circleci.com/pipelines/github/rsimha/amphtml/149/workflows/ae42212b-07e7-4c91-b7fa-f5fbaf7fb6b3/jobs/548/parallel-runs/0/steps/0-107)

**Reference:** https://mochajs.org/api/tutorial-custom-reporter.html

### CODE REVIEW

File: build-system/tasks/e2e/mocha-dots-reporter.js [hunk 1]
```diff
@@ -15,52 +15,64 @@
...
+        process.stdout.write(green(icon.success));
+        wrapIfNecessary();
+      })
+      .on(EVENT_TEST_FAIL, () => {
+        process.stdout.write(red(icon.failure));
+        wrapIfNecessary();
+      })
+      .on(EVENT_TEST_PENDING, () => {
+        process.stdout.write(yellow(icon.ignore));
+        wrapIfNecessary();
```
Comment (danielrozenberg) [inline build-system/tasks/e2e/mocha-dots-reporter.js:63]: These three callbacks are basically the same, why not move the `process.stdout.write` part into the `wrapIfNecessary` function (+ rename it to something like `printDot`)

File: build-system/tasks/e2e/mocha-dots-reporter.js [hunk 2]
```diff
@@ -15,52 +15,64 @@
...
+      })
+      .once(EVENT_RUN_END, () => {
+        Base.list(this.failures);
+        const {passes, pending, failures, tests} = runner.stats;
+        console.log(
+          `Executed ${failures + passes} of ${tests}`,
+          `(Skipped ${pending})`,
+          failures == 0 ? green('SUCCESS') : red(`${failures} FAILED`)
```
Comment (danielrozenberg) [inline build-system/tasks/e2e/mocha-dots-reporter.js:71]: Supernit: move the SUCCESS/FAILED print to the start of the line, this is how every other test reporting platform I've used displays its output 🙃

**Review by danielrozenberg — APPROVED** [review]

File: build-system/tasks/e2e/mocha-dots-reporter.js (hunk 2 above)
Comment (rsimha) [reply build-system/tasks/e2e/mocha-dots-reporter.js:69]: The super-dots reporter that we use with Karma prints the status at the end of the line.

image

File: build-system/tasks/e2e/mocha-dots-reporter.js (hunk 1 above)
Comment (rsimha) [reply build-system/tasks/e2e/mocha-dots-reporter.js]: Good idea. Done.

**Pull request merged**